src/code/z_skelanime.c

```diff
@@ -798,245 +795,299 @@ void SkelAnime_InterpFrameTable(s32 limbCount, Vec3s* dst, Vec3s* start, Vec3s*
     }
 }
 
+static u32 sDisabledTransformTaskGroups = 0;
+static u32 sCurAnimTaskGroup;
+
 /**
- * zeroes out the current request count
+ * Resets the AnimTaskQueue.
+ * Zeroing the task count will stop the queue from processing tasks until new ones are added.
```

Comment:
Nit: I'd just remove this sentence, it made me doubt whether the old tasks were "really" gone. Maybe `Resets the AnimTaskQueue.` -> `Removes all tasks from the AnimTaskQueue.`

Reply:
The same reaction you have to my comment, I have to yours 😅. If the comment says "removes tasks", but i just see it zeroing a count, I would be confused how that is "removing" a task. Which is why I am explaining what zeroing the task count really does.

Reply:
If this comment was reworded to this would it help:
"Because the queue only processes up to `count` number of tasks, zeroing `count` will stop the queue from processing old entries, effectively resetting the whole queue."

Reply:
idk that still makes it seem like the rest of the array beyond `count` has valid tasks instead of being garbage, it implies that the old entries still exist and the word "effectively" makes it seem like an accident.

I'm kind of struggling to communicate my point here. Basically I think code comments should be high-level, saying what the code *means* instead of what the code *does* (since if you want to know what it does you can read the code). For example, if I was implementing a vector data structure I would not want comment it like this:

```c++
class IntVector {
 public:
  // Sets the count to 0.
  IntVector() {
    count = 0;
  }
   
  // Writes a new element to the array and adds 1 to the count.
  void push(int x) {
    data[count] = x;
    count++;
  }

  // Subtracts 1 from the count, effectively preventing any user from
  // processing the element that was at the back of the array before.
  void pop() {
    count--;
  }

  // Sets the count to 0, effectively preventing any user from processing
  // old entries until new ones are added.
  void clear() {
    count = 0;
  }
   
 private:
  int count;
  int data[MAX_SIZE];
}
```

Instead I would comment it like this:

```c++
class IntVector {
 public:
  // Creates an empty vector.
  IntVector() {
    count = 0;
  }
   
  // Appends an element to the end of the vector.
  void push(int x) {
    data[count] = x;
    count++;
  }

  // Removes an element from the end of the vector.
  void pop() {
    count--;
  }

  // Removes all elements from the vector.
  void clear() {
    count = 0;
  }
   
 private:
  int count;
  int data[MAX_SIZE];
}
```
Here the comments hit at what's going on conceptually, so the user can build a mental model of how to use the data structure, instead of saying how the code "happens" to work.

Reply:
Im not implying that anything is an accident. "Effectively" here just means that doing this one action is what leads to something else occurring, this is its effect.
I give the high level explanation in the first sentence. I think its also fine to explain how the action taken leads to that desired high level effect. To me, the wording as-is is not a problem.

Someone else can weigh in if they'd like to see this changed as well

Reply:
I agree with cadmic's points

I can suggest the following wording:
```
Clear the current task queue. The discarded tasks will then not be processed.
```

Reply:
When reading this code for the first time, I had to think for a moment how just zeroing the count leads to the queue being "reset" or tasks being discarded. Just saying the tasks are "discarded and wont be processed" doesn't help with my intent.
But point taken that the wording should change.

Reply:
maybe some version of my second sentence makes more sense as a comment in-line in the function? And the suggested comment can remain as the high-level function comment?

I do feel somewhat strongly about explaining this somewhere. It may seem obvious to you guys how it works, but it took me enough thought to feel like someone else could use the explanation as well.

Reply:
Oh fwiw it seems straightforward to me
Or at least similar if not identical to a gazillion other queue/dynamic-array/buffer implementations

If you want to elaborate on how it works I'd put a comment inside the function (though I don't think it's necessary)

Reply:
alright i will concede and just not explain it